### ISSUE TITLE
Android: One element labelling another

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30846]
## Requires API Proposal
This issue requires a new API. An API proposal should be added and discussed before proceeding with implementation. The API proposal can be added in the comments of this issue or linked as a separate issue.

## Description
No concept of "labelFor" or "labeledBy" or any way to pass a reference to another element as the accessibilityLabel property of an element. Used most when building complex forms.

## React Native version:
v0.63

## Expected Behavior
Android has a labelFor property on View, which allows you to pass a reference to the ID of another view, that this view should act as the label for. When Focusing on the other view, it will announce <label> for <other view's label>.

For example, if you have these two Views
```
<Text labelFor="123">Wifi</Text>
<Checkbox accessibilityLabel="checkbox" id="123" />
```
And you focus on the checkbox, it would announce "checkbox for Wifi".

## Android Details
Any API to add this feature should manipulate the underlying View's labelFor property.  It will need a reference of the ID of the element that is being labelled to properly work.

**Comment by kacieb** [comment]
There is an existing `testID` property, I wonder if that could be used for this. I'm not sure though as I'm not sure the underlying implementation.

**Comment by blavalla** [comment]
See the comment here on one potential approach for implementation:

https://github.com/facebook/react-native/issues/26739#issuecomment-819149373
